(Docs) Update choco package doc (#2316)
puppetlabs/bolt: documentation/automating_windows_targets.md

# Automating Windows targets

> Here are some common tasks that you can accomplish with Bolt on Windows targets.

Reviewer: ```suggestion
Here are some common use cases that you can accomplish with Bolt on Windows targets.
```
'Tasks' just feels like kind of a loaded word, but there's not a lot of great synonyms :P

Author: I agree. I could not for the life of me think of a synonym. Use cases is perfect

these steps:

1.  Run your PowerShell script on a Windows target.

> 2.  Create an inventory file to store information about the target.

Reviewer: ? I like the `1.` because you don't have to modify the other items if we add a new step in the middle of the list

Author: Oh, this is a VS Code extension doing this when I save. I also prefer just using 1. I'll see if I can disable this functionality.

Author: Found the setting that was doing this 😉

      Add-BoltModule -Module puppetlabs-chocolatey
      ```

   - If you want to create a project (named `choco_project`) that includes the Chocolatey module. Create a directory named `choco_project` and run the following command inside the directory:
    
      _\*nix shell command_

      ```shell
      bolt project init chocho_project --modules puppetlabs-chocolatey
      ```

      _PowerShell cmdlet_

      ```powershell
      New-BoltProject -Name choco_project -Modules puppetlabs-chocolatey
      ```

1. Install Chocolatey on your Windows target using the `apply` command:
   
   _\*nix shell command_

    ```shell
    bolt apply -e 'include chocolatey' -t <TARGET URI> -u <USER> -p <PASSWORD> --transport winrm
    ```

    _PowerShell cmdlet_

    ```powershell
    Invoke-BoltApply -Execute "include chocolatey"  -Targets <TARGET URI> -User <USER> -Password <PASSWORD> -Transport winrm
    ```

1. Use the built-in Package task to install Frogsay on your target:
    
    _\*nix shell command_

    ```shell
    bolt task run package -t <TARGET URI> -u <USER> -p <PASSWORD> --transport winrm action=install name=frogsay
    ```

    _PowerShell cmdlet_

    ```powershell

>     Invoke-BoltTask -Name package -Targets <TARGET URI> -User <USER> -Password <PASSWORD> -Transport winrm -Params @{action='install'} @{name='frogsay'}

Reviewer 2: ```suggestion
    Invoke-BoltTask -Name package -Targets <TARGET URI> -User <USER> -Password <PASSWORD> -Transport winrm action='install' name='frogsay'
```

Elsewhere in the docs we show passing params in the same manner as *nix shells, so I think it'd be good to maintain consistency here.

Author: @beechtom Does it still need the quotes? ie. `action='install'`

Reviewer 2: Not if it's a single word, no.

    that you specified in your inventory file. For more information, see
    [Specify targets](running_bolt_commands.md#adding-options-to-bolt-commands).

## Deploy a package with Bolt and Chocolatey

You can use Bolt with Chocolatey to deploy a package on a Windows target. First,
use the `apply` command to install Chocolatey on the target. Next, use Puppet's
Chocolatey package provider to install the package.

This example installs the Frogsay package on a Windows target.

**Before you begin:**
- [Install Bolt](bolt_installing_modules.md)
- Configure Windows Remote Management (WinRM) on your Windows target.

To install a package with Chocolatey:
1. Install Chocolatey on your Windows target using the `apply` command:
   _\*nix shell command_

    ```shell

>     bolt apply -e 'include chocolatey' -t <TARGET URI> -u <USER> -p <PASSWORD> --transport winrm --no-ssl

Reviewer: ```suggestion
    bolt apply -e 'include chocolatey' -t <TARGET URI> -u <USER> -p <PASSWORD> --transport winrm
```
We've gotten complaints before about encouraging turning ssl off :P I don't particularly care either way, but we should at lest have the powershell command and nix one match

Author: Ah cool ty. Sidenote: I actually assumed it's off by default with Powershell because of the way -Ssl is described in the cmdlets reference. Is that true or should we look into rewording that?

Reviewer: They should both default to `true` https://puppet.com/docs/bolt/latest/bolt_transports_reference.html#ssl

    ```

2. Use the built-in Package task to install Frogsay on your target:
   ```shell
    bolt task run package -t <TARGET URI> -u <USER> -p <PASSWORD> --transport winrm --no-ssl action=install name=frogsay
    ```

    _PowerShell cmdlet_

    ```powershell
    Invoke-BoltTask -Name package -Targets <TARGET URI> -User <USER> -Password <PASSWORD> -Transport winrm -Params @{action='install'} @{name='frogsay'}
    ```     

3. Run `frogsay` on your target to test:

   ```shell
   bolt command run 'frogsay ribbit' -t <TARGET URI> -u <USER> -p <PASSWORD> --transport winrm --no-ssl
   ```

   _PowerShell cmdlet_

   ```powershell
   Invoke-BoltCommand 'frogsay ribbit' -Targets <TARGET URI> -User <USER> -Password <PASSWORD> -Transport winrm
   ``` 
  
   Your output should look something like this:
   ```shell
   Started on example.windowstarget.net...
   Finished on example.windowstarget.net:
      STDOUT:

                WORRIED ABOUT LONG LINES? FROG CAN HOLD YOUR PLACE FOR UP TO 65534
                SECONDS BEFORE IT FORGETS WHAT IT'S DOING AND HOPS AWAY.
                /

>           @..@

Reviewer: OMG I LOVE THIS
